src/test/java/seedu/address/model/event/TimeTest.java

```diff
@@ -12,7 +13,31 @@
 
 class TimeTest {
 
-    // @Rishit add isValidTime checks here
+    @Test
+    public void isValidTimeTest() {
+        // null time
+        assertThrows(NullPointerException.class, () -> Time.isValidTime(null));
+
+        // invalid time
+        assertFalse(Time.isValidTime("")); // empty string
+        assertFalse(Time.isValidTime(" ")); // spaces only
+        assertFalse(Time.isValidTime("meeting*")); // contains not time
+        assertFalse(Time.isValidTime("meeting*")); // contains not time
+        assertFalse(Time.isValidTime("meeting*")); // contains not time
+        assertFalse(Time.isValidTime("April, 12, 2024 - 10:00 AM")); // no day
+        assertFalse(Time.isValidTime("Friday, 12, 2024 - 10:00 AM")); // no month
+        assertFalse(Time.isValidTime("Friday, April,2024 - 10:00 AM")); //no day (int)
+        assertFalse(Time.isValidTime("Friday, April, 12 - 10:00 AM")); // no year
+        assertFalse(Time.isValidTime("Friday, April, 12, 2024 AM")); // no time
+        assertFalse(Time.isValidTime("Friday, April, 12, 2024 - 10:00")); // no am/pm
+        assertFalse(Time.isValidTime("Friday, April, 12, 2024 -10:00 AM")); //missing space
+        assertFalse(Time.isValidTime("Friday, April, 12, 2024  10:00 AM")); //missing hyphen
+        assertFalse(Time.isValidTime("Friday, April 12 2024 - 10:00 AM")); //missing comma
+
+        assertTrue(Time.isValidTime("Friday, April, 12, 2024 - 10:00 AM"));
+        assertTrue(Time.isValidTime("Thursday, April, 04, 2024 - 10:00 AM"));
+        assertTrue(Time.isValidTime("Thursday, May, 02, 2024 - 09:00 AM"));
```

Comment:
Might be better to separate the assertTrue and assertFalse tests into separate methods